src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/DynamicExpressionUtilities.cs

```diff
@@ -1,30 +1,48 @@
-﻿using Microsoft.DotNet.Interactive.Http;
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
 using System;
-using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
-using System.Linq;
-using System.Text;
 using System.Text.RegularExpressions;
 using System.Threading;
-using System.Threading.Tasks;
 
 namespace Microsoft.DotNet.Interactive.Http.Parsing
 {
 #nullable enable
     internal static class DynamicExpressionUtilites
     {
-
-        const string DateTime = "$datetime";
-        const string LocalDateTime = "$localDatetime";
+        const string DateTimeMacroName = "$datetime";
+        const string LocalDateTimeMacroName = "$localDatetime";
         const string OffsetRegex = """(?:\s+(?<offset>[-+]?[^\s]+)\s+(?<option>[^\s]+))?""";
         const string TypeRegex = """(?:\s+(?<type>rfc1123|iso8601|'.+'|".+"))?""";
 
         internal static Regex guidPattern = new Regex(@$"^\$guid$", RegexOptions.Compiled);
-        internal static Regex dateTimePattern = new Regex(@$"^\{DateTime}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
-        internal static Regex localDateTimePattern = new Regex(@$"^\{LocalDateTime}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
+        internal static Regex dateTimePattern = new Regex(@$"^\{DateTimeMacroName}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
+        internal static Regex localDateTimePattern = new Regex(@$"^\{LocalDateTimeMacroName}{TypeRegex}{OffsetRegex}$", RegexOptions.Compiled);
         internal static Regex randomIntPattern = new Regex(@$"^\$randomInt(?:\s+(?<arguments>-?[^\s]+)){{0,2}}$", RegexOptions.Compiled);
-        internal static Regex timestampPattern = new Regex($@"^\$timestamp{OffsetRegex}$", RegexOptions.Compiled);
+        internal static Regex timestampPattern = new Regex($@"^\$timestamp{OffsetRegex}$", RegexOptions.Compiled);
+
+        private static Func<bool, DateTimeOffset> GetDateTimeFunc = DefaultGetDateTime;

```

Comment:
A custom delegate type would help reveal the intent of the `bool` parameter without having to resort to comments.

Reply:
Will update.